### PR TITLE
PLNSRVCE-1310: address concerns with duration metrics

### PR DESCRIPTION
**Pull request by gabemontero**
I've marked this as a WIP PR, but that is not because I do not think this PR is ready for review.  

Rather, it is possible we drop some of the commits of this PR, based on decisions we make during review.

I  think the commit descriptions are self explanatory wrt what is changed here, but to help ensure clarity, this PR:
- switches to controller-runtime based prometheus integration wrt setting up the service, signal handling, and metric registration
- a commit that reduces metric labelling / cardinality (though I note where comparable metrics upstream and pipelinerun level cardinality)
- a commit that switches from gauge to histogram
- various TODO comments around the need for further examination of metrics, from both implementations details, and overlap with upstream metrics

### CODE REVIEW

**Comment by gabemontero** [comment]
I've also added a switch to more "controller runtime friendly" argument and logger libraries (i.e. the ones commonly used in the github.com/redhat-appstudio/*-service controllers ).

File: collector/collector.go [hunk 1]
```diff
@@ -17,36 +17,45 @@
...
 
 func NewCollector(logger log.Logger, client client.Client) (*PipelineServiceCollector, error) {
 	//TODO should this be converted to a Desc so we can use constant metrics
-	durationScheduled := prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "pipelinerun_duration_scheduled_seconds",
-		Help: "Duration in seconds for a PipelineRun to be scheduled.",
-		//TODO the uid will get us a unique entry, but wouldn't adding namespace will help usability when consuming this metric?
-		// also, labels at an individual PipelineRun is a scalability issue with prometheus
-	}, []string{"name", "uid"})
+	// shipwright establishment metrics buckets is []float64{0, 1, 2, 3, 5, 7, 10, 15, 20, 30}
```
Comment (gabemontero) [inline collector/collector.go:43]: will remove before we merge - just wanted to provide some context on the bucket settings from other projects that I took as a starting point

File: collector/collector.go [hunk 2]
```diff
@@ -17,36 +17,45 @@
...
+	// shipwright ramp up metrics buckets is prometheus.LinearBuckets(0, 1, 10)
```
Comment (gabemontero) [inline collector/collector.go:44]: will remove before we merge - just wanted to provide some context on the bucket settings from other projects that I took as a starting point

File: collector/collector.go [hunk 3]
```diff
@@ -17,36 +17,45 @@
...
+		Buckets: prometheus.LinearBuckets(0, 1, 10),
+	}, []string{"namespace"})
 
 	//TODO should this be converted to a Desc so we can use constant metrics
-	durationCompleted := prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "pipelinerun_duration_completed_seconds",
-		Help: "Duration in seconds for a PipelineRun to complete.",
-		//TODO the uid will get us a unique entry, but wouldn't adding namespace help usability with humans consuming this metric?
-	}, []string{"name", "uid"})
+	// shipwright completion duration buckets is prometheus.LinearBuckets(50, 50, 10)
```
Comment (gabemontero) [inline collector/collector.go:52]: will remove before we merge - just wanted to provide some context on the bucket settings from other projects that I took as a starting point

File: collector/collector.go [hunk 4]
```diff
@@ -17,36 +17,45 @@
...
+	//TODO how does this compare to the upstream tekton metric tekton_pipelines_controller_pipelinerun_duration_seconds_[bucket, sum, count] that is of type Histogram/LastValue(Gauge) and has pipelinerun name level cardinality
```
Comment (gabemontero) [inline collector/collector.go:53]: have started looking at this and the 2 metrics are working similarly - need to investigate how configurable the tekton / knative histograms are wrt bucket config

**Comment by gabemontero** [comment]
Looks like the need for https://issues.redhat.com/browse/PLNSRVCE-1330 has manifested itself in the failed PaC pull request job in this repo @Roming22 @adambkaplan @enarha 

from https://github.com/openshift-pipelines/pipeline-service-exporter/pull/25/checks?check_run_id=13941062129 


`failed to create task run pod "pipeline-service-metrics-exporter-pull-request-lvxgh-init": pods "pipeline-service-metrics-exporter-pull-request-lvxgh-init-pod" is forbidden: unable to validate against any security context constraint: [provider "anyuid": Forbidden: not usable by user or serviceaccount, spec.volumes[10]: Invalid value: "csi": csi volumes are not allowed to be used, provider "restricted": Forbidden: not usable by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable by user or serviceaccount, provider "pcap-dedicated-admins": Forbidden: not usable by user or serviceaccount, provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount, provider "log-collector-scc": Forbidden: not usable by user or serviceaccount, provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount, provider "hostnetwork-v2": Forbidden: not usable by user or serviceaccount, provider "hostnetwork": Forbidden: not usable by user or serviceaccount, provider "hostaccess": Forbidden: not usable by user or serviceaccount, provider "splunkforwarder": Forbidden: not usable by user or serviceaccount, provider "node-exporter": Forbidden: not usable by user or serviceaccount, provider "privileged": Forbidden: not usable by user or serviceaccount]. Maybe missing or invalid Task bnr-tenant/init`

that is because out PaC job is referencing older tasks in the build-definition task catalog, specifically the use of shared resource csi driver, that are no longer supported.

**Comment by gabemontero** [comment]
Aside from ^^, given the slack discussions and the use of histograms gaining traction for us, I'm removing the WIP

**Comment by gabemontero** [comment]
ok @sbose78 has created the pipeline-service-exporter image in quay.io/redhat-appstudio, but we need to merge a PR to really create another PUSH jobs (at least I'm not skilled enough to do that)

So I'll see about getting this merged, then see what happens.

**Review by Roming22 — APPROVED** [review]

**Pull request merged**